src/core/config.c

```diff
@@ -0,0 +1,108 @@
+#include "config.h"
+#include "xdpw.h"
+#include "logger.h"
+
+#include <dictionary.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <iniparser.h>
+
+void print_config(enum LOGLEVEL loglevel, struct xdpw_config *config) {
+	logprint(loglevel, "config: Outputname  %s",config->screencast_conf.output_name);
+}
+
+// NOTE: calling finish_config won't prepare the config to be read again from config file
+// with init_config since to pointers and other values won't be reset to NULL, or 0
+void finish_config(struct xdpw_config *config) {
+	logprint(DEBUG, "config: destroying config");
+
+	// screencast
+	free(&config->screencast_conf.output_name);
+}
+
+static void getstring_from_conffile(dictionary *d, char *key, char **dest, char *fallback) {
+	if (*dest != NULL) {
+		return;
+	}
+	const char *c = iniparser_getstring(d, key, fallback);
+	if (c == NULL) {
+		return;
+	}
+	// Allow keys without value as default
+	if (strcmp(c, "") != 0) {
+		*dest = strdup(c);
+	} else {
+		*dest = fallback ? strdup(fallback) : NULL;
+	}
+}
+
+static bool file_exists(const char *path) {
+	return path && access(path, R_OK) != -1;
+}
+
+static char *config_path(char *prefix, char *filename) {
+	if (!prefix || !prefix[0] || !filename || !filename[0]) {
+		return NULL;
+	}
+
+	char *config_folder = "xdg-desktop-portal-wlr";
+
+	size_t size = 3 + strlen(prefix) + strlen(config_folder) + strlen(filename);
+	char *path = calloc(size, sizeof(char));
+	snprintf(path, size, "%s/%s/%s", prefix, config_folder, filename);
+	return path;
+}
+
+static void config_parse_file(char *configfile, struct xdpw_config *config) {
+	dictionary *d = iniparser_load(configfile);
+
+	// screencast
+	getstring_from_conffile(d, "screencast:output_name", &config->screencast_conf.output_name, NULL);
+
+	iniparser_freedict(d);
+	logprint(DEBUG, "config: config file parsed");
+	print_config(DEBUG, config);
+}
+
+static char *get_config_path(void) {
+	const char *home = getenv("HOME");
+	char *config_home_fallback = calloc(1 + strlen(home) + strlen("/.config"), sizeof(char));
+	snprintf(config_home_fallback, strlen(config_home_fallback), "%s/.config", home);
```

Comment:
This doesn't work since strlen(config_home_fallback) is 0.

Reply:
Forgot the second snprintf. Thanks for finding!